redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/provision/handler/UpdateDatabaseServerRegistrationHandler.java

```diff
@@ -75,11 +93,96 @@ public void accept(Event<UpdateDatabaseServerRegistrationRequest> event) {
             dbServerConfig.setConnectionUserName(updatedUserName);
 
             databaseServerConfigService.update(dbServerConfig);
+
+            CloudContext cloudContext = request.getCloudContext();
+            syncSslCertificateIfNeeded(cloudContext, request.getCloudCredential(), dbStack, request.getDatabaseStack());
+
             response = new UpdateDatabaseServerRegistrationSuccess(request.getResourceId());
+            LOGGER.debug("Database server registration update successfully finished for {}", cloudContext);
         } catch (Exception e) {
             response = new UpdateDatabaseServerRegistrationFailed(request.getResourceId(), e);
             LOGGER.warn("Error updating the database server registration:", e);
         }
-        eventBus.notify(response.selector(), new Event<>(event.getHeaders(), response));
+        return response;
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<UpdateDatabaseServerRegistrationRequest> event) {
+        UpdateDatabaseServerRegistrationFailed failure = new UpdateDatabaseServerRegistrationFailed(resourceId, e);
+        LOGGER.warn("Error updating the database server registration:", e);
+        return failure;
+    }
+
+    private void syncSslCertificateIfNeeded(CloudContext cloudContext, CloudCredential cloudCredential, DBStack dbStack, DatabaseStack databaseStack)
```

Comment:
The SSL cert syncer logic could be isolated into an individual service.

Reply:
Good point, has been moved to the new class `DatabaseServerSslCertificateSyncService` and covered in `DatabaseServerSslCertificateSyncServiceTest`.